LoopKit/SettingsStore.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  SettingsStore.swift
+//  LoopKit
+//
+//  Created by Darin Krauss on 10/14/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import HealthKit
+
+public protocol SettingsStoreDelegate: AnyObject {
+    
+    /**
+     Informs the delegate that the settings store has updated settings data.
+     
+     - Parameter settingsStore: The settings store that has updated settings data.
+     */
+    func settingsStoreHasUpdatedSettingsData(_ settingsStore: SettingsStore)
+    
+}
+
+public class SettingsStore {
+    
+    public weak var delegate: SettingsStoreDelegate?
+    
+    private let lock = UnfairLock()
+    
+    private let userDefaults: UserDefaults?
+    
+    private var settings: [Int64: StoredSettings]
```

Comment:
No need for bit width specification.  We'll eventually need to persist this, of course, but that is covered in https://tidepool.atlassian.net/browse/LOOP-884, looks like.

Reply:
Will change `Int64` in all anchors and related to `Int`.

Reply:
Referencing above comment:

"Actually, it is an Int64 because that matches the type of the field in each Core Data object. I can change the type of the field Core Data Swift object to be Int, but the actual data model requires a bit width so there'd have to be some type conversion somewhere along the way.

I will leave it as-is unless you feel strongly about it. If you want it changed, please let me know at what place you'd like the type conversion."